kubernetes/resource_kubernetes_service.go

```diff
@@ -140,6 +158,31 @@ func resourceKubernetesServiceCreate(d *schema.ResourceData, meta interface{}) e
 	log.Printf("[INFO] Submitted new service: %#v", out)
 	d.SetId(buildId(out.ObjectMeta))
 
+	if out.Spec.Type == api.ServiceTypeLoadBalancer {
+		log.Printf("[DEBUG] Waiting for load balancer to assign IP/hostname")
+
+		err = resource.Retry(5*time.Minute, func() *resource.RetryError {
+			svc, err := conn.CoreV1().Services(out.Namespace).Get(out.Name, meta_v1.GetOptions{})
+			if err != nil {
+				log.Printf("[DEBUG] Received error: %#v", err)
+				return resource.NonRetryableError(err)
+			}
+
+			lbIngress := svc.Status.LoadBalancer.Ingress
```

Comment:
As discussed via Slack - in case the service suddenly doesn't exist it would get caught by the err check above, if the creation of LB failed then we'd time out after 5 mins. Exposing the reason of failed creation in the timeout error message is a job for a separate PR.